From 6868c4ab2f00efcaea93b09cf7b4f7d544656d10 Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@qt.io>
Date: Mon, 6 Oct 2025 14:42:57 +0800
Subject: [PATCH 127/239] QQuickPane: disconnect signals in destructor

It seems that the private connection isn't automatically disconnected
in time to avoid a sporadic crash when destroying Pane for native
styles.

Fixes: QTBUG-140838
Pick-to: 6.8
Change-Id: I01e24b33d532d4c818f7a1dd9da7c03bcae503fd
Reviewed-by: Oliver Eftevaag <oliver.eftevaag@qt.io>
(cherry picked from commit ba56f1e4638f9752903d587129b0474e845ce416)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quicktemplates/qquickpane.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/quicktemplates/qquickpane.cpp b/src/quicktemplates/qquickpane.cpp
index 25099ef350..2cc5c6375e 100644
--- a/src/quicktemplates/qquickpane.cpp
+++ b/src/quicktemplates/qquickpane.cpp
@@ -266,6 +266,12 @@ QQuickPane::QQuickPane(QQuickItem *parent)
 QQuickPane::~QQuickPane()
 {
     Q_D(QQuickPane);
+    if (d->contentItem) {
+        // It's possible for the focus frame to be removed as a child of our contentItem
+        // upon our destruction, so disconnect to avoid getting our slot getting called during this.
+        QObjectPrivate::disconnect(d->contentItem, &QQuickItem::childrenChanged,
+            d, &QQuickPanePrivate::contentChildrenChange);
+    }
     d->removeImplicitSizeListener(d->contentItem);
     d->removeImplicitSizeListener(d->firstChild);
 }
-- 
2.51.2

